pkg/ccl/backupccl/backup_job.go

```diff
@@ -319,11 +327,26 @@ func backup(
 		}
 	}
 
-	resumerSpan.RecordStructured(&types.StringValue{Value: "writing backup manifest"})
+	// Write a `BACKUP_MANIFEST` file to support backups in mixed-version clusters
+	// with 22.2 nodes.
+	//
+	// TODO(adityamaru): We can stop writing `BACKUP_MANIFEST` in 23.2
+	// because a mixed-version cluster with 23.1 nodes will read the
+	// `BACKUP_METADATA` instead.
 	if err := backupinfo.WriteBackupManifest(ctx, defaultStore, backupbase.BackupManifestName,
 		encryption, &kmsEnv, backupManifest); err != nil {
 		return roachpb.RowCount{}, err
 	}
+
+	// Write a `BACKUP_METADATA` file along with SSTs for all the alloc heavy
+	// fields elided from the `BACKUP_MANIFEST`.
+	if backupinfo.WriteMetadataWithFilesSST.Get(&settings.SV) {
```

Comment:
Clean backport, only new diff is this cluster setting that defaults to false.

Reply:
I decided to make the setting a metamorphic bool to ensure that we have coverage for both the old-style and new-style backup manifests in our tests.